From ef89b02221130b604ebb574c89306b0bfd44c9b7 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Tue, 16 Nov 2021 17:06:21 +0100
Subject: [PATCH 258/268] Increment reference count when restoring reference

Otherwise the count will be wrong after an out of memory failure in
reinterpretAsFormat.

Pick-to: 6.2 5.15
Fixes: QTBUG-98377
Change-Id: Ice51d47a6db9277126a5c7337e14aaf5ddee3a10
Reviewed-by: Eirik Aavitsland <eirik.aavitsland@qt.io>
(cherry picked from commit 1a8b7eb1d4f27e74621ee94c01dbeda3afd302c7)
---
 src/gui/image/qimage.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/image/qimage.cpp b/src/gui/image/qimage.cpp
index 5560b7a3bb..b482bf8227 100644
--- a/src/gui/image/qimage.cpp
+++ b/src/gui/image/qimage.cpp
@@ -2255,6 +2255,7 @@ bool QImage::reinterpretAsFormat(Format format)
         // In case detach() ran out of memory
         if (!d) {
             d = oldD;
+            d->ref.ref();
             return false;
         }
     }
-- 
2.34.1

